service/awsconfig/v8/resource/lifecycle/current.go

```diff
@@ -95,34 +101,212 @@ func (r *Resource) GetCurrentState(ctx context.Context, obj interface{}) (interf
 
 		if len(instances) == 0 {
 			r.logger.LogCtx(ctx, "level", "debug", "message", fmt.Sprintf("did not find the guest cluster instances being in state '%s'", autoscaling.LifecycleStateTerminatingWait))
-			return nil, nil
-
 		} else {
 			r.logger.LogCtx(ctx, "level", "debug", "message", fmt.Sprintf("found %d guest cluster instances being in state '%s'", len(instances), autoscaling.LifecycleStateTerminatingWait))
 		}
 	}
 
-	// TODO check if NodeConfig exists
-	// TODO create NodeConfig if it does not exist
-	// TODO check if NodeConfig has final state
+	{
+		for _, instance := range instances {
+			r.logger.LogCtx(ctx, "level", "debug", "message", "looking for node config for the guest cluster")
+
+			privateDNS, err := r.privateDNSForInstance(*instance.InstanceId)
+			if err != nil {
+				return nil, microerror.Mask(err)
+			}
+
+			n := v1.NamespaceDefault
+			o := metav1.GetOptions{}
+
+			_, err = r.g8sClient.CoreV1alpha1().NodeConfigs(n).Get(privateDNS, o)
+			if errors.IsNotFound(err) {
+				r.logger.LogCtx(ctx, "level", "debug", "message", "did not find node config for guest cluster node")
+
+				err := r.createNodeConfig(ctx, customObject, *instance.InstanceId, privateDNS)
```

Comment:
This block here is only to create the node config. Further inspection of the node config status is done in further reconciliation loops at runtime. After the next resync period we will see the node config exists and the cleanup can happen below.